homeassistant/components/influxdb/__init__.py

```diff
@@ -218,7 +218,7 @@ def event_to_json(event: Event) -> dict[str, Any] | None:
         state: State | None = event.data.get(EVENT_NEW_STATE)
         if (
             state is None
-            or state.state in (STATE_UNKNOWN, "", STATE_UNAVAILABLE)
+            or state.state in (STATE_UNKNOWN, "", STATE_UNAVAILABLE, None)
```

Comment:
This would be a bug in an integration if it happens. `None` should always be converted to `STATE_UNKNOWN` by the base Entity class before being set in the state machine.